providers/implementations/signature/ecdsa_sig.c

```diff
@@ -155,32 +274,33 @@ static int ecdsa_signverify_init(void *vctx, void *ec,
     ctx->operation = operation;
 
     OSSL_FIPS_IND_SET_APPROVED(ctx)
-    if (!ecdsa_set_ctx_params(ctx, params))
+    if (!set_ctx_params(ctx, params))
         return 0;
 #ifdef FIPS_MODULE
     if (!ossl_fips_ind_ec_key_check(OSSL_FIPS_IND_GET(ctx),
                                     OSSL_FIPS_IND_SETTABLE0, ctx->libctx,
                                     EC_KEY_get0_group(ctx->ec), desc,
-                                    operation == EVP_PKEY_OP_SIGN))
+                                    (operation & (EVP_PKEY_OP_SIGN
+                                                  | EVP_PKEY_OP_SIGNMSG)) != 0))
         return 0;
 #endif
     return 1;
 }
 
 static int ecdsa_sign_init(void *vctx, void *ec, const OSSL_PARAM params[])
 {
-    return ecdsa_signverify_init(vctx, ec, params, EVP_PKEY_OP_SIGN,
-                                 "ECDSA Sign Init");
-}
-
-static int ecdsa_verify_init(void *vctx, void *ec, const OSSL_PARAM params[])
-{
-    return ecdsa_signverify_init(vctx, ec, params, EVP_PKEY_OP_VERIFY,
-                                 "ECDSA Verify Init");
+    return ecdsa_signverify_init(vctx, ec, ecdsa_set_ctx_params, params,
+                                 EVP_PKEY_OP_SIGN, "ECDSA Sign Init");
 }
 
-static int ecdsa_sign(void *vctx, unsigned char *sig, size_t *siglen,
-                      size_t sigsize, const unsigned char *tbs, size_t tbslen)
+/*
+ * Sign tbs without digesting it first.  This is suitable for "primitive"
+ * signing and signing the digest of a message, i.e. should be used with
+ * implementations of the keytype related algorithms.
```

Comment:
What does 'keytype related algorithms' mean?

Reply:
The EVP_SIGNATURE algorithm that have the same name as the keytype it uses.  I.e. "RSA", as opposed to "RSA-SHA256".  Not sure how to word this more properly.

Reply:
Of course, this could be named specifically in the comment instead of being this vague...  Will do

Reply:
...  or cutting away the last part of the sentence.  That's what I ended up doing.